src/base/bittorrent/torrenthandle.cpp

```diff
@@ -416,6 +416,17 @@ QVector<TrackerEntry> TorrentHandle::trackers() const
     return entries;
 }
 
+bool TorrentHandle::tracker(const QString &trackerUrl, TrackerEntry &entry) const
+{
+    for (const auto &tracker : m_nativeHandle.trackers()) {
```

Comment:
We could, but there is no benifit. It would look like this, more code, arguably less readable, and will generate more bloated machine code, just because we have capturing lambda, calling it for every object, more variables on stack and so on. 

``` diff
diff --git a/src/base/bittorrent/torrenthandle.cpp b/src/base/bittorrent/torrenthandle.cpp
index 7669efca2..55db91f4f 100644
--- a/src/base/bittorrent/torrenthandle.cpp
+++ b/src/base/bittorrent/torrenthandle.cpp
@@ -418,13 +418,16 @@ QVector<TrackerEntry> TorrentHandle::trackers() const
 
 bool TorrentHandle::tracker(const QString &trackerUrl, TrackerEntry &entry) const
 {
-    for (const auto &tracker : m_nativeHandle.trackers()) {
-        if (QString::fromStdString(tracker.url) == trackerUrl) {
-            entry = tracker;
-            return true;
-        }
-    }
-    return false;
+    const std::vector<lt::announce_entry> nativeTrackers = m_nativeHandle.trackers();
+    const auto iter = std::find_if(nativeTrackers.cbegin(), nativeTrackers.cend(), [&trackerUrl](const lt::announce_entry &tracker)
+    {
+        return (QString::fromStdString(tracker.url) == trackerUrl);
+    });
+    if (iter == nativeTrackers.cend())
+        return false;
+
+    entry = *iter;
+    return true;
 }
```